libstorage.apib

```diff
@@ -468,33 +531,50 @@ Internal server error
 
     + Schema
 
-            { "$ref": "https://raw.githubusercontent.com/emccode/libstorage/master/libstorage.json#/definitions/internalServerError" }
+            { "$ref": "https://raw.githubusercontent.com/codedellemc/libstorage/master/libstorage.json#/definitions/internalServerError" }
 
 ## Get with Attachments [GET /volumes?{attachments}]
 Gets a list of Volume resources and their attached resources for all configured
 services.
 
 + Parameters
 
-    + attachments (required)
-
-        A flag that indicates that volume attachments should be retrieved.
-        If this flag is provided then the `LibStorage-InstanceID` header must
-        be set -- a comma-delimited list of service name/instance ID,
-        colon-delimited pairs.
+    + attachments (string,optional)
```

Comment:
Hi @mhrabovcin,

Because query string params are all either string or boolean. Additionally I wanted the param to be backwards compatible when it was a boolean flag, so if a truthy value is passed I can interpret it on the backend like so:

```go
// ParseVolumeAttachmentTypes parses a value into a VolumeAttachmentsTypes
// value.
func ParseVolumeAttachmentTypes(v interface{}) VolumeAttachmentsTypes {
	switch tv := v.(type) {
	case VolumeAttachmentsTypes:
		return tv
	case bool:
		if tv {
			return VolumeAttachmentsTypes(14)
		} else {
			return VolumeAttachmentsNone
		}
	case int:
		return VolumeAttachmentsTypes(tv)
	case uint:
		return VolumeAttachmentsTypes(tv)
	case int8:
		return VolumeAttachmentsTypes(tv)
	case uint8:
		return VolumeAttachmentsTypes(tv)
	case int16:
		return VolumeAttachmentsTypes(tv)
	case uint16:
		return VolumeAttachmentsTypes(tv)
	case int32:
		return VolumeAttachmentsTypes(tv)
	case uint32:
		return VolumeAttachmentsTypes(tv)
	case int64:
		return VolumeAttachmentsTypes(tv)
	case uint64:
		return VolumeAttachmentsTypes(tv)
	case string:
		if i, err := strconv.ParseInt(tv, 10, 64); err == nil {
			return ParseVolumeAttachmentTypes(i)
		}
		if b, err := strconv.ParseBool(tv); err == nil {
			return ParseVolumeAttachmentTypes(b)
		}
		return VolumeAttachmentsNone
	}
	return VolumeAttachmentsNone
}
```